### DISCUSSION TITLE
How to set a thumbnail for custom video (e.g Youtube, Loom)

### DISCUSSION THREAD

**Discussion by gagandeepgill** [igordanchenko/yet-another-react-lightbox#143] [Q&A] [answered]
Is there a way to set a thumbnail that is not of type `video`?

For example here we are using https://github.com/igordanchenko/yet-another-react-lightbox/issues/49#issuecomment-1505689829 a youtube video, how would you set a `poster` prop on this?

**Comment by igordanchenko** [comment, marked as answer]
Yes, of course. You just need to define custom `render.thumbnail` function to render thumbnails for custom slides.

```jsx
<Lightbox
  slides={[
    {
      type: "youtube",
      src: "https://www.youtube.com/embed/aqz-KE-bpKQ",
      thumbnail: "https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/images/BigBuckBunny.jpg",
      title: "Big Buck Bunny",
      width: 2160,
      height: 1215,
    },
  ]}
  plugins={[Thumbnails]}
  render={{
    slide: ({ slide, rect }) => // render custom slide
    thumbnail: ({ slide, ...rest }) =>
      slide.type === "youtube" ? (
        <ImageSlide slide={{ src: slide.thumbnail }} {...rest} />
      ) : undefined
  }}
  // ...
/>
```

https://codesandbox.io/p/sandbox/yet-another-react-lightbox-143-gznp29?file=%2Fsrc%2FApp.tsx
